cmd/node-agent/main.go

```diff
@@ -0,0 +1,302 @@
+	desiredConfigAnnotation = "node-agent.alpha.coreos.com/desired-config"
+	currentConfigAnnotation = "node-agent.alpha.coreos.com/current-config"
+	kubeletVersionKey       = "KUBELET_VERSION"
+	kubeletConfigKey        = "KUBELET_CONFIG"
+	kubeletFlagsKey         = "KUBELET_OPTS"
+	kubeletEnvPath          = "/etc/kubernetes/kubelet.env"
+	kubeletService          = "kubelet.service"
+	configMapFlagsKey       = "kubelet-flags"
+)
+
+var (
+	sysdConn       *dbus.Conn
+	client         clientset.Interface
+	configMapStore cache.Store
+
+	myPodName = os.Getenv(myPodNameEnvName)
+)
+
+func main() {
+	if myPodName == "" {
+		glog.Fatalf("%s env var not set.", myPodNameEnvName)
+	}
+	run()
+}
+
+func run() {
+	client = newAPIClient()
+	nodename, err := getNodeName()
+	if err != nil {
+		glog.Fatal(err)
+	}
+
+	glog.Infof("starting node agent, watching node: %s", nodename)
+
+	sysdConn, err = dbus.New()
+	if err != nil {
+		glog.Fatal(err)
+	}
+
+	nodeopts := api.ListOptions{
+		FieldSelector: fields.ParseSelectorOrDie(fmt.Sprintf("metadata.name=%s", nodename)),
+	}
+	_, nodeController := framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(lo api.ListOptions) (runtime.Object, error) {
+				return client.Core().Nodes().List(nodeopts)
+			},
+			WatchFunc: func(lo api.ListOptions) (watch.Interface, error) {
+				return client.Core().Nodes().Watch(nodeopts)
+			},
+		},
+		&v1.Node{},
+		30*time.Second,
+		framework.ResourceEventHandlerFuncs{
+			UpdateFunc: nodeUpdateCallback,
+		},
+	)
+	var configMapController *framework.Controller
+	configMapStore, configMapController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(lo api.ListOptions) (runtime.Object, error) {
+				return client.Core().ConfigMaps(api.NamespaceSystem).List(lo)
+			},
+			WatchFunc: func(lo api.ListOptions) (watch.Interface, error) {
+				return client.Core().ConfigMaps(api.NamespaceSystem).Watch(lo)
+			},
+		},
+		&v1.ConfigMap{},
+		30*time.Second,
+		framework.ResourceEventHandlerFuncs{},
+	)
+
+	go configMapController.Run(wait.NeverStop)
+	nodeController.Run(wait.NeverStop)
+}
+
+func newAPIClient() clientset.Interface {
+	config, err := restclient.InClusterConfig()
+	if err != nil {
+		glog.Fatal(err)
+	}
+	return clientset.NewForConfigOrDie(config)
+}
+
+func nodeUpdateCallback(_, newObj interface{}) {
+	glog.Infof("begin node update callback")
+
+	newNode, ok := newObj.(*v1.Node)
+	if !ok {
+		glog.Errorf("received unexpected type: %T (expected *v1.Node)", newObj)
+		return
+	}
+	onDiskConfig, err := parseKubeletEnvFile(kubeletEnvPath)
+	if err != nil {
+		glog.Error(err)
+		return
+	}
+	if newNode.Annotations == nil {
+		newNode.Annotations = make(map[string]string)
+	}
+	rawconf, ok := newNode.Annotations[desiredConfigAnnotation]
+	if !ok {
+		if _, ok := newNode.Annotations[currentConfigAnnotation]; !ok {
+			setCurrentConfigAnnotation(newNode, onDiskConfig)
+		}
+		glog.Info("no configuration annotation found, ignoring")
```

Comment:
nit: `("no %s annotation found for node %s, ignoring", desiredConfigAnnotation, newNode.Name)`